### ISSUE TITLE
Improve tolerance of :uri parsing to connect

### ISSUE THREAD

**Issue by reify-james-vickers** [protojure/lib#134]
Currently `protojure.grpc.client.providers.http2/connect` requires a very specific format for `:uri` and does not accept some that it perhaps could.  The URI's it seems to accept have scheme as well as port number, e.g. `http://localhost:8080`.  Any piece missing makes the parsing fail, sometimes with a relatively opaque exception.  It would be helpful for instance to support formats like:

- `http://some-host` or `https://some-host`, using the default port numbers for these protocols (i.e. `80` for `http` and `443` for `https`)
  - Currently omitting the port number causes an exception: `(NumberFormatException) at java.lang.Integer/parseInt (Integer.java:614): null`
- `some-host.com` or similar URI's that omit the connection protocol; not clear if it's really necessary to have the protocol or if it can default to something like `https` or correspond to what `:ssl` is defined/provided as.

**Comment by ghaskins** [comment]
Yeah, this has bugged me for a while, too.  Just never got around to fixing it.  PR welcome.
